### PR TITLE
clang-tidy modernize pass

### PR DESCRIPTION
**Pull request by est77**

### CODE REVIEW

File: src/appleseedmaya/utils.h [hunk 1]
```diff
@@ -55,12 +55,12 @@ class MStatus;
 class NonCopyable
 {
   protected:
-    NonCopyable() {}
-    ~NonCopyable() {}
+    NonCopyable() = default;
```
Comment (dictoon) [inline src/appleseedmaya/utils.h:58]: Why not use `foundation::NonCopyable` here?

**Comment by dictoon** [comment]
Cool!

Can we try clang-tidy on appleseed core?

**Comment by est77** [comment]
Yes, we can. It's actually easy to do. some references:
https://www.kdab.com/clang-tidy-part-1-modernize-source-code-using-c11c14/
https://baptiste-wicht.com/posts/2017/03/clang-tidy-static-analysis-integration-in-sonarqube.html
https://github.com/OpenImageIO/oiio/pull/1649

**Comment by dictoon** [comment]
Cool, then we should probably do it on appleseed at some point. Adding `override` and converting 0 to `nullptr` seems like a good start.

**Pull request merged**